### ISSUE TITLE
dask tests suddenly fail

### ISSUE THREAD

**Issue by lgray** [scikit-hep/fastjet#228]
https://github.com/scikit-hep/fastjet/blob/main/tests/test_008-dask.py#L99

No longer passes as of this week.
It passed on Friday with 2.2.3 with no problem?

@agoose77 @jpivarski 

```
E           TypeError: __class__ assignment: 'MomentumArray4D' object layout differs from 'Array'
E           
E           This error occurred while calling
E           
E               ak.zip(
E                   arrays = {'pt': <Array-typetracer [...] type='## * var * float32'>, '...
E                   depth_limit = None
E                   parameters = None
E                   with_name = 'Momentum4D'
E                   right_broadcast = False
E                   optiontype_outside_record = False
E                   highlevel = True
E                   behavior = {('*', 'Vector2D'): <class 'vector.backends.awkward.Vector...
E               )
```

**Comment by agoose77** [comment]
@lgray this is concerning — we're _also_ seeing this in our CI in `main`, overnight. I can't reproduce this locally. 

Will keep you posted RE diagnosis.

**Comment by agoose77** [comment]
OK, I'm now able to reproduce this on my machine. You can drop the troubleshooting, I'll take figure it out on our end!

**Comment by agoose77** [comment]
This PR should fix it. https://github.com/scikit-hep/awkward/pull/2534

**Comment by lgray** [comment]
fixed with awkward 2.2.4
